include/pisa/v1/README.md

```diff
@@ -0,0 +1,79 @@
+> This document is a **work in progress**.
+
+# Introduction
+
+In our efforts to come up with the v1.0 of both PISA and our index format,
+we should start a discussion about the shape of things from the point of view
+of both the binary format and how we can use it in our library.
+
+## Index Format specification
+
+This document mainly discusses the binary file format of each index component,
+as well as how these components come together to form a cohesive structure.
+
+## Reference Implementation
+
+Along with format description and discussion, this directory includes some
+reference implementation of the discussed structures and some algorithms working on them.
+
+The goal of this is to show how things work on certain examples,
+and find out what works and what doesn't and still needs to be thought through.
+
+> Look in `test/test_v1.cpp` for code examples.
+
+# Posting Files
+
+> Example: `v1/raw_cursor.hpp`.
+
+Each _posting file_ contains a list of blocks of data, each related to a single term,
+preceded by a header encoding information about the type of payload.
+
+> Do we need the header? I would say "yes" because even if we store the information
+> somewhere else, then we might want to (1) verify that we are reading what we think
+> we are reading, and (2) verify format version compatibility.
+> The latter should be further discussed.
+
+```
+Posting File := Header, [Posting Block]
```

Comment:
Is there one `Header` for each `Posting Block`?

Reply:
No, one header, followed by a list of blocks.